### ISSUE TITLE
Outdated message for failed minikube ISO download

### ISSUE THREAD

**Issue by larrikinventures** [kubernetes/minikube#4039]
Thanks for k8s!

The related issue link in the console output for `minikube start` is outdated and sends me to a 404 - https://github.com/kubernetes/minikube/issue/3922

As an aside, the download always hangs at around 1%. I'm not sure if it's some kind of timeout, but it's always a different percentage, roughly around a MB into the download. I can download https://storage.googleapis.com/minikube/iso/minikube-v1.0.0.iso from Firefox just fine, but the command line download is super choppy (as in, download for 10s, hang for 1 minute, download for 10s, hang for 1 minute, repeat). I ended up manually copying the downloaded file from Firefox into the cache folder which worked. Note - I'm in Australia with a slow connection.

minikube version: v1.0.0
OSX 10.13

```bash
> minikube start
😄  minikube v1.0.0 on darwin (amd64)
🤹  Downloading Kubernetes v1.14.0 images in the background ...
🔥  Creating hyperkit VM (CPUs=2, Memory=2048MB, Disk=20000MB) ...
💿  Downloading Minikube ISO ...
 1.44 MB / 142.88 MB [>-----------------------------------------]   1.01% 26m20s

💣  Unable to start VM
❌  Error:         [ISO_DOWNLOAD_FAILED] unable to cache ISO: https://storage.googleapis.com/minikube/iso/minikube-v1.0.0.iso: failed to download: failed to download to temp file: failed to copy contents: read tcp 192.168.1.4:54524->216.58.203.112:443: read: connection reset by peer
💡  Advice:        A firewall is likely blocking minikube from reaching the internet. You may need to configure minikube to use a proxy.
📘  Documentation: https://github.com/kubernetes/minikube/blob/master/docs/http_proxy.md
⁉️   Related issues:
    ▪ https://github.com/kubernetes/minikube/issue/3922

😿  If the above advice does not help, please let us know:
👉  https://github.com/kubernetes/minikube/issues/new
```

**Comment by manish1635** [comment]
You are behind a firewall. Please use this link to set proxy:
https://github.com/kubernetes/minikube/blob/master/docs/http_proxy.md

**Comment by larrikinventures** [comment]
Unfortunately I'm not. The console output already linked me there but there is no appropriate proxy - I'm just at home, connected via my router, no firewall configured. As I said the connection _is_ slow, but the odd hanging at intervals is unique to the ISO download.

Anyway, this ticket is for the bad link in that console output. Not sure if it's supposed to point somewhere else or should just be removed.

**Comment by manish1635** [comment]
I hope you already tried these:
set HTTP_PROXY=null
set HTTPS_PROXY=null
set NO_PROXY=localhost,127.0.0.1,10.96.0.0/12,192.168.99.1/24,192.168.39.0/24

If not works, Then not sure about your broadband configuration. However, you could check with your provider to give you some proxy IP.

**Comment by jjfraney** [comment]
The docker-env variables apply to the docker engine and the proxy environment variables ares needed by the engine to pull docker images. 

https://storage.googleapis.com/minikube/iso/minikube-v1.0.0.iso is NOT a docker image.  Its the virtual machine boot image.  Docker engine is not pulling it.  minikube downloads it.  Does minikube observe these environment variables when it downloads the boot image?  I don't think it does.  I expect minikube reads proxy environment from the shell environment (or powershell context).

My corporate proxy works for me.  I can download stuff.  So this is not a network issue.  When minikube attempts to download the virtual machine's boot image, it is NOT using the proxy, I think, even with these --docker-env vars correctly set.

**Comment by alexism** [comment]
I'm running into a similar issue.
The iso is reachable with `curl`:
```
 curl --head https://storage.googleapis.com/minikube/iso/minikube-v1.0.0.iso
HTTP/2 200
x-guploader-uploadid: AEnB2UqfKv3-bh1dYYau-KaHcK3f5hhWhkBqYFymDdRj0RZrpcKCL6VClid3-7GveKjQh7wGvKB8_tMqNS2ECffvGY_8h2vO4Q
expires: Tue, 09 Apr 2019 23:45:24 GMT
date: Tue, 09 Apr 2019 22:45:24 GMT
cache-control: public, max-age=3600
last-modified: Tue, 26 Mar 2019 03:04:41 GMT
etag: "24ea5719eddc8699f4308610192cefc2"
x-goog-generation: 1553569481916933
x-goog-metageneration: 1
x-goog-stored-content-encoding: identity
x-goog-stored-content-length: 149821440
content-type: application/x-iso9660-image
x-goog-hash: crc32c=GEF68Q==
x-goog-hash: md5=JOpXGe3chpn0MIYQGSzvwg==
x-goog-storage-class: STANDARD
accept-ranges: bytes
content-length: 149821440
server: UploadServer
alt-svc: quic=":443"; ma=2592000; v="46,44,43,39"
```

Minikube output:
```
$ minikube start
😄  minikube v1.0.0 on darwin (amd64)
🤹  Downloading Kubernetes v1.14.0 images in the background ...
2019/04/09 15:36:56 No matching credentials were found, falling back on anonymous
2019/04/09 15:36:56 No matching credentials were found, falling back on anonymous
2019/04/09 15:36:56 No matching credentials were found, falling back on anonymous
2019/04/09 15:36:56 No matching credentials were found, falling back on anonymous
2019/04/09 15:36:56 No matching credentials were found, falling back on anonymous
2019/04/09 15:36:56 No matching credentials were found, falling back on anonymous
2019/04/09 15:36:56 No matching credentials were found, falling back on anonymous
2019/04/09 15:36:56 No matching credentials were found, falling back on anonymous
2019/04/09 15:36:56 No matching credentials were found, falling back on anonymous
2019/04/09 15:36:56 No matching credentials were found, falling back on anonymous
2019/04/09 15:36:56 No matching credentials were found, falling back on anonymous
2019/04/09 15:36:56 No matching credentials were found, falling back on anonymous
2019/04/09 15:36:56 No matching credentials were found, falling back on anonymous
🔥  Creating virtualbox VM (CPUs=2, Memory=2048MB, Disk=20000MB) ...
💿  Downloading Minikube ISO ...

💣  Unable to start VM
❌  Error:         [ISO_DOWNLOAD_FAILED] unable to cache ISO: https://storage.googleapis.com/minikube/iso/minikube-v1.0.0.iso: failed to download: failed to download to temp file: download failed: 5 error(s) occurred:

* Temporary download error: Get https://storage.googleapis.com/minikube/iso/minikube-v1.0.0.iso: dial tcp: lookup storage.googleapis.com on 17.128.100.9:53: lame referral
* Temporary download error: Get https://storage.googleapis.com/minikube/iso/minikube-v1.0.0.iso: dial tcp: lookup storage.googleapis.com on 17.128.100.9:53: lame referral
* Temporary download error: Get https://storage.googleapis.com/minikube/iso/minikube-v1.0.0.iso: dial tcp: lookup storage.googleapis.com on 17.128.100.9:53: lame referral
* Temporary download error: Get https://storage.googleapis.com/minikube/iso/minikube-v1.0.0.iso: dial tcp: lookup storage.googleapis.com on 17.128.100.9:53: lame referral
* Temporary download error: Get https://storage.googleapis.com/minikube/iso/minikube-v1.0.0.iso: dial tcp: lookup storage.googleapis.com on 17.128.100.9:53: lame referral
💡  Advice:        A firewall is likely blocking minikube from reaching the internet. You may need to configure minikube to use a proxy.
📘  Documentation: https://github.com/kubernetes/minikube/blob/master/docs/http_proxy.md
⁉️   Related issues:
    ▪ https://github.com/kubernetes/minikube/issue/3922

😿  If the above advice does not help, please let us know:
👉  https://github.com/kubernetes/minikube/issues/new
```

**Comment by wayne-t** [comment]
encountered similar problem here. I am able to download from chrome or using wget though.

**Comment by wayne-t** [comment]
For people who might have the same problem, here is reference where to put downloaded files to `cache`. I found it's useful since I had more missing files other than .iso mentioned above. 
https://github.com/kubernetes/minikube/blob/master/docs/offline.md

**Comment by jjfraney** [comment]
@wayne-t thanks for that link.

For minikube to use your corporate proxy, I discovered it necessary to set proxy for the minikube program and for the docker containers it runs.  Both of these are required.  My fail was not knowing how to do the former in powershell, but its similar to bash:
`$env:HTTP_PROXY="......"`
`$env:HTTPS_PROXY="....."`
`$env:NO_PROXY="......"`
`minikube-windows-amd64.exe start --vm-driver hyperv --docker-env=http_proxy=$Env:HTTP_PROXY --docker-env=https_proxy=$Env:HTTPS_PROXY --docker-env=no_proxy=$Env:NO_PROXY  --hyperv-virtual-switch="Default Switch"`

I have a followon problem, which I'll enter on a different issue.

**Comment by tstromberg** [comment]
This should be fixed in v1.0.1. Thanks for reporting the issue!

**Comment by irtaza909** [comment]
> For people who might have the same problem, here is reference for where to put downloaded files to `cache`. I found it's useful since I had more missing files other than .iso mentioned above.
> https://github.com/kubernetes/minikube/blob/master/docs/offline.md

minikube start problem please help. i have even downloaded the iso file manually and move it in using "mv minikube-v1.7.0.iso /home/irtaza/.minikube/cache/iso" even then it is stuck on " Downloading VM boot image ..."
